sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,211 @@
+
+## Motivation
+
+Currently, KubeVirt does not support modifying or reporting the link state of virtual network interfaces.
+This limitation can lead to challenges in implementing advanced networking scenarios, such as:
+- Simulating link failures for testing and debugging.
+- Managing multi-network setups, where some interfaces need to be intentionally disabled.
+
+## Goals
+
+1. Allow users to configure the link state (up/down) of individual interfaces through the KubeVirt API.
+2. Enable link state configuration at both VM creation and runtime.
+3. Report the current domain link state of each VM interface.
+
+## Non Goals
+
+1. Affect network plumbing on the host - the setup process will remain unchanged.
+2. Report the link state of the underlying network provider.
+3. Report the link state of interfaces inside the guest operating system.
+
+## User Stories
+
+As a VM owner I would like to be able to:
+
+- Start the VM with one or more interfaces with their link set to `down`.
+- Toggle the link state of one or more interfaces of a running VM.
+- Hot plug an interface with a link state set to `down`.
+- Control the link state of interfaces connected to primary and secondary networks.
+
+- I want my interfaces link states to persist following a migration.
+- I want the domain link state to be reported per VMI.
+
+## Repos
+
+kubevirt/kubevirt.
+
+## Design
+
+### API Addition
+Currently, the `Interface` struct has the `State` field (currently used for hot-unplug):
+```go
+type Interface struct {
+    ...
+    State InterfaceState `json:"state,omitempty"`
+    ...
+}
+```
+
+Type `InterfaceState` currently has a single allowed value:  
+```go
+type InterfaceState string
+
+const (
+	InterfaceStateAbsent InterfaceState = "absent"
+)
+```
+
+The following values will be added to specify the required link state:
+```go
+const (
+    InterfaceStateLinkUp   InterfaceState = "up"
+    InterfaceStateLinkDown InterfaceState = "down"
+)
+```
+
+An empty value will be considered as `up`.
+
+The network validator will be adjusted to allow the two new values for all interfaces regardless of whether they are 
+connected to primary or secondary networks, or whether they use a core binding or a binding plugin.
+
+> [!IMPORTANT]
+> Controlling the link state of SR-IOV NICs will not be supported as we cannot control their link state via the domain.
+
+virt-launcher's `Converter` component will be adjusted to take the interface State field into account when creating a new domain XML.
+In case the field's value is `InterfaceStateLinkDown`, the interface will be created as follows:
+```xml
+<interface>
+  ...
+  <link state='down'/>
+  ...
+</interface>
+```
+
+> [!NOTE]
+> For additional details please see libvirt's [documentation](https://libvirt.org/formatdomain.html#modifying-virtual-link-state).
+
+virt-launcher's `LibvirtDomainManager.SyncVMI` will be extended to support updating the interfaces link state while the VM is running.
```

Comment:
It's not entirely clear to me how this process will work. Could you please elaborate and explain the flow?